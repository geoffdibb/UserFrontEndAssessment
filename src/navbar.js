import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import Register from "./Register";
import ViewUsers from "./ViewUsers";
import Home from "./Home";
import Login from "./Login";

export class Navbar extends Component {

    constructor() {
        super();
        this.state = {
            data: []
        };
    }

    render() {
        return (



            <div>
                <h1 backgroundcolor="black">Home
                  </h1>
                <Router>
                    <button>
                        <Link to="/Home">Home</Link>
                    </button>
                    <button>
                        <Link to="/Register">RegisterAccount</Link>
                    </button>
                    <button>
                        <Link to="/Login">Login</Link>
                    </button>
                    <button>
                        <Link to="/ViewUsers">View All Accounts</Link>
                    </button>
                    <Route path="/Home" render={(props) => <Home passedFunction={this.onLoad} dataSent={this.state.data} />} />

                    <Route path="/Register" render={(props) => <Register passedFunction={this.onLoad} dataSent={this.state.data} />} />
                    <Route path="/Login" render={(props) => <Login passedFunction={this.onLoad} dataSent={this.state.data} />} />
                    <Route path="/ViewUsers" render={(props) => <ViewUsers passedFunction={this.onLoad} dataSent={this.state.data} />} />



                </Router>
            </div>
        )
    }
}
export default Navbar;