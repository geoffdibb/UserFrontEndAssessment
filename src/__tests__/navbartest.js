import React from 'react';
import ReactDOM from 'react-dom';
import navbar from '../navbar';
import renderer from 'react-test-renderer';


it('snapshot navbar', () => {
  const tree = renderer.create(<navbar/>).toJSON()
  expect(tree).toMatchSnapshot();
})