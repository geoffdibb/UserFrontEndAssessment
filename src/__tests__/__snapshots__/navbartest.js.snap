// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot navbar 1`] = `<navbar />`;
